docs/runbooks/Container-README.md

```diff
@@ -0,0 +1,281 @@
+# Refactor Coaching & Mentoring Platform with Docker & Docker Compose
+
+*This project is a Rust-based backend/web API that connects to a PostgreSQL database. It uses Docker and Docker Compose for easy local development and deployment, and includes utilities for database management, migrations, and more. You can choose to run PostgreSQL either locally (via Docker) or remotely by configuring the environment variables.*
+
+## Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- [Docker](https://www.docker.com/products/docker-desktop) (version 20+)
+- [Docker Compose](https://docs.docker.com/compose/install/) (version 1.29+)
+
+## Project Setup
+
+1. **Clone the repository**:
+
+   ```bash
+   git clone <repository-url>
+   cd <repository-directory>
+   ```
+
+2. **Environment Configuration**:
+
+   You'll need to decide whether you're connecting to a **local PostgreSQL container** (using Docker) or a **remote PostgreSQL instance** (e.g., on a different host or in the cloud). This is configured using `.env` files.
+
+   - **For local PostgreSQL** (default, Docker-based):
+     Create a `.env.local` file based on the template provided below and specify `POSTGRES_HOST=postgres`.
+
+   - **For remote PostgreSQL**:
+     Create a `.env.remote-db` file and set `POSTGRES_HOST` to the external IP or hostname of the remote PostgreSQL instance.
+
+   Example `.env.local` for local development:
+  
+   ```env
+   POSTGRES_USER=refactor
+   POSTGRES_PASSWORD=password
+   POSTGRES_DB=refactor_platform
+   POSTGRES_HOST=postgres
+   POSTGRES_PORT=5432
+   WEB_PORT=8080
+   USERNAME=appuser
+   USER_UID=1000
+   USER_GID=1000
+   ```
+
+   Example `.env.remote-db` for remote PostgreSQL:
+
+   ```env
+   POSTGRES_USER=remote_user
+   POSTGRES_PASSWORD=remote_password
+   POSTGRES_DB=remote_db
+   POSTGRES_HOST=remote-db-host.com
+   POSTGRES_PORT=5432
+   WEB_PORT=8080
+   USERNAME=remote_appuser
+   USER_UID=1001
+   USER_GID=1001
+   ```
+
+3. **Review the `docker-compose.yaml` file**:
+
+   The Docker Compose file is configured to use environment variables defined in your `.env` files. The PostgreSQL container can either run locally (if specified in your environment file) or you can connect to a remote instance by setting the appropriate `POSTGRES_HOST`.
+
+   ```yaml
+   services:
+     postgres:
+       image: postgres:13
```

Comment:
@lmcdonough can we try moving up to the latest stable version of Postresql, which is 17.x? I've been testing using 14.x for development but I think it'd be worth deploying with the latest stable version.

@calebbourg what do you think?

Reply:
I'm all for latest versions

Reply:
done.